### PR TITLE
Prevent invalid version creation for default and empty versions

### PR DESCRIPTION
**Pull request by mkleen**
This prevents invalid version creation for the case where the version
id is either empty or handled as default value from the Setting.

## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/VersionTest.java [hunk 1]
```diff
@@ -70,4 +73,11 @@ public void test_can_parse_next_major_release_version() {
             Version.CURRENT.luceneVersion.major + 1, 0, 0)
         ));
     }
+
+    @Test
+    public void test_empty_version() {
+        Version emptyVersion = IndexMetadata.SETTING_INDEX_VERSION_CREATED.get(Settings.EMPTY);
```
Comment (mkleen) [inline server/src/test/java/org/elasticsearch/VersionTest.java:79]: This led to -3 without the adjustment.

File: server/src/test/java/org/elasticsearch/common/settings/SettingTests.java [hunk 2]
```diff
@@ -1011,4 +1014,10 @@ public void testAffixMapUpdateWithNullSettingValue() {
         assertEquals("", value);
     }
 
+    @Test
+    public void test_version_default_setting() {
+        var versionSetting = Setting.versionSetting(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT);
```
Comment (mkleen) [inline server/src/test/java/org/elasticsearch/common/settings/SettingTests.java:1019]: This led to 1.5 without the adjustment.

File: server/src/main/java/org/elasticsearch/common/settings/Setting.java [hunk 3]
```diff
@@ -1084,7 +1084,7 @@ public void apply(T value, Settings current, Settings previous) {
     }
 
     public static Setting<Version> versionSetting(final String key, final Version defaultValue, Property... properties) {
-        return new Setting<>(key, s -> Integer.toString(defaultValue.externalId), s -> Version.fromId(Integer.parseInt(s)), properties);
+        return new Setting<>(key, s -> Integer.toString(defaultValue.internalId), s -> Version.fromId(Integer.parseInt(s)), properties);
```
Comment (mkleen) [inline server/src/main/java/org/elasticsearch/common/settings/Setting.java:1087]: Version lookup is based on the internal id not the external id See https://github.com/crate/crate/blob/master/server/src/main/java/org/elasticsearch/Version.java#L141

File: extensions/jmx-monitoring/src/test/java/io/crate/beans/NodeInfoTest.java [hunk 4]
```diff
@@ -93,7 +94,8 @@ public void setup() {
                                                        ShardRoutingState.UNASSIGNED));
 
         var routingTable = RoutingTable.builder().add(indexRoutingTableBuilder).build();
-        var meta = IndexMetadata.builder(tableName).numberOfShards(1).numberOfReplicas(2);
+        var settings = Settings.builder().put(IndexMetadata.SETTING_INDEX_VERSION_CREATED.getKey(), Version.CURRENT);
+        var meta = IndexMetadata.builder(tableName).settings(settings).numberOfShards(1).numberOfReplicas(2);
```
Comment (seut) [inline extensions/jmx-monitoring/src/test/java/io/crate/beans/NodeInfoTest.java:98]:
```suggestion
        var meta = IndexMetadata.builder(tableName).settings(settings(Version.CURRENT)).numberOfShards(1).numberOfReplicas(2);
```

File: extensions/jmx-monitoring/src/test/java/io/crate/beans/NodeInfoTest.java [hunk 5]
```diff
@@ -224,7 +226,8 @@ public void test_partitioned_tables() {
 
 
         var routingTable = RoutingTable.builder().add(indexRoutingTableBuilder).build();
-        var meta = IndexMetadata.builder(tableName).numberOfShards(1).numberOfReplicas(2);
+        var settings = Settings.builder().put(IndexMetadata.SETTING_INDEX_VERSION_CREATED.getKey(), Version.CURRENT);
+        var meta = IndexMetadata.builder(tableName).settings(settings).numberOfShards(1).numberOfReplicas(2);
```
Comment (seut) [inline extensions/jmx-monitoring/src/test/java/io/crate/beans/NodeInfoTest.java:230]:
```suggestion
        var meta = IndexMetadata.builder(tableName).settings(settings(Version.CURRENT)).numberOfShards(1).numberOfReplicas(2);
```

File: server/src/test/java/org/elasticsearch/common/settings/SettingTests.java [hunk 6]
```diff
@@ -1011,4 +1013,15 @@ public void testAffixMapUpdateWithNullSettingValue() {
         assertEquals("", value);
     }
 
+    @Test
+    public void test_empty_version_from_setting() {
+        var emptyVersion = IndexMetadata.SETTING_INDEX_VERSION_CREATED.get(Settings.EMPTY);
+        assertThat(emptyVersion, is((Version.V_EMPTY)));
+    }
+
+    @Test
```
Comment (seut) [inline server/src/test/java/org/elasticsearch/common/settings/SettingTests.java:1022]: Would be good to add a comment here that we had a regression where the defaultValue was not handled correctly

**Review by seut — COMMENTED** [review]
Good catch, added some suggestions.

File: extensions/jmx-monitoring/src/test/java/io/crate/beans/NodeInfoTest.java [hunk 7]
```diff
@@ -37,6 +37,7 @@
 import org.elasticsearch.cluster.routing.RoutingTable;
 import org.elasticsearch.cluster.routing.ShardRoutingState;
 import org.elasticsearch.cluster.routing.TestShardRouting;
+import org.elasticsearch.common.settings.Settings;
```
Comment (seut) [inline extensions/jmx-monitoring/src/test/java/io/crate/beans/NodeInfoTest.java:40]: unused now?

**Review by seut — APPROVED** [review]
👍

**Pull request merged**